modules/soc_psa/psasoc.py

```diff
@@ -93,8 +98,23 @@
 f.write(str(responestatus))
 f.close()
 vin_list = json.loads(responsetext)
-vin_id  = vin_list ['_embedded']['vehicles'][0]['id']
-vin_vin  = vin_list ['_embedded']['vehicles'][0]['vin']
+
+# if VIN is not given, first car will be selected:
+if not vehicleId:
+    vin_id = vin_list ['_embedded']['vehicles'][0]['id']
+    vin_vin = vin_list ['_embedded']['vehicles'][0]['vin']
+
+else:
+    # search for entry of given VIN
+    vin_selected = filter(lambda x: x['vin'] == vehicleId, vin_list['_embedded']['vehicles'])
+    vin_id = vin_selected [0]['id']
+    vin_vin = vin_selected [0]['vin']
```

Comment:
Konntest Du das testen? (Leider funktioniert der Link auf das Forum nicht.)
Aus dem Bauch heraus würde ich vermuten, dass das filter-Ergebnis etwa so aussieht: `vin_selected = {"id": 1, "vin":xy}`. Dann wäre:
```suggestion
    vin_id = vin_selected['id']
```
`vin_vin` ist auch kein besonders aussagekräftiger Name. Gilt nicht sowieso `vin==vin_vin`?

Reply:
Moin,

ja, mit meinem Account und Auto hat es erfolreich funktioniert.
Der Index ist notwendig, da nur Elemente aus dem Array herausgefiltert werden, aber nicht die Struktur selber verändert wird. Das führt dazu, dass es als Ergebnis immer ein Array als Ergebnis zurück gibt: Entweder es ist leer (Filter nicht gefunden) oder in unserem Fall der gesuchte Eintrag mit Index 0. Siehe auch weiter unten bei dem alten Batterie-Filter von mrinas.

In `psareply2filterVINlp` steht die richtige VIN und ID.

Die Variablennamen finde ich auch nicht klasse, die waren aber so. 
Wenn Du magst, refactor ich das.

Reply:
In Python 2 geht das so, in Python 3 geht das nicht. Es könnte eine Überlegung wert sein das ganze gleich Python 3-kompatibel zu schreiben. Irgendwann wird das jemand konvertieren müssen (für oWB 2) und das könnte die nötige Änderung schon vorwegnehmen.

```python
vehicle_selected = next(vehicle for vehicle in vin_list['_embedded']['vehicles'] if vehicle['vin'] == vehicleId)
```
müsste meine ich sowohl in Python 2 als auch in 3 gehen und dann ist das Ergebnis auch so wie LKuemmel sagt, also `[0]` nicht nötig.

Oder auch:
```python
vehicles = vin_list['_embedded']['vehicles']
vehicle_selected = next(vehicle for vehicle in vehicles if vehicle['vin'] == vehicleId) if vehicleId else vehicles[0]
```
Dann hättest du sogar das ganze if/else auch noch mit erschlagen.

Reply:
Ich habe das Modul mal etwas refactored... 
* Variablen umbenannt
* Filter umgestellt